src/node_http_parser.cc

```diff
@@ -462,14 +458,11 @@ class Parser : public AsyncWrap, public StreamListener {
     if (!cb->IsFunction())
       return 0;
 
-    // We came from consumed stream
-    if (current_buffer_.IsEmpty()) {
-      // Make sure Buffer will be in parent HandleScope
-      current_buffer_ = scope.Escape(Buffer::Copy(
-          env()->isolate(),
-          current_buffer_data_,
-          current_buffer_len_).ToLocalChecked());
-    }
+    // Make sure Buffer will be in parent HandleScope
+    Local<Object> current_buffer_ = scope.Escape(Buffer::Copy(
+        env()->isolate(),
+        current_buffer_data_,
+        current_buffer_len_).ToLocalChecked());
```

Comment:
Suggestion: since you're creating a new buffer now, just restrict it to at..length:
```suggestion
    // Make sure Buffer will be in parent HandleScope
    Local<Object> current_buffer = scope.Escape(
        Buffer::Copy(env()->isolate(), at, length));
```
Update the MakeCallback call to just pass a single argument and update parserOnBody in lib/_http_common.js to get rid of the slice() call (and the extraneous function parameters.) That should alleviate the performance hit.

https://github.com/nodejs/node/blob/47ea59fad1ea97ffeab7b670f45b99943ce9543b/lib/_http_common.js#L129

Minor style issue: the local should be called `current_buffer` (no final underscore, that's to distinguish data class members.)